### PR TITLE
colbert query scorers

### PR DESCRIPTION
**Pull request by IvanPleshkov**
Implement Metric scorer for Colbert. It's required for raw scorer 

### All Submissions:

* [ ] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [ ] Have you followed the guidelines in our Contributing document?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [ ] Does your submission pass tests?
2. [ ] Have you formatted your code locally using `cargo +nightly fmt --all` command prior to submission?
3. [ ] Have you checked your code using `cargo clippy --all --all-features` command?

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: lib/segment/src/vector_storage/query_scorer/mod.rs [hunk 1]
```diff
@@ -11,3 +16,14 @@ pub trait QueryScorer<TVector: ?Sized> {
...
     fn score_internal(&self, point_a: PointOffsetType, point_b: PointOffsetType) -> ScoreType;
 }
+
+pub fn score_multi<TMetric: Metric>(a: &MultiDenseVector, b: &MultiDenseVector) -> ScoreType {
+    let mut sum = 0.0;
+    for a in a.iter() {
+        sum += b
+            .iter()
+            .map(|b| TMetric::similarity(a, b))
+            .fold(0.0, |a, b| if a > b { a } else { b });
```
Comment (agourlay) [inline lib/segment/src/vector_storage/query_scorer/mod.rs:26]: I find it a bit confusing to have the accumulator named `a` and `b` like the elements of the vectors in the loop.

File: lib/segment/src/vector_storage/query_scorer/mod.rs [hunk 2]
```diff
@@ -11,3 +16,14 @@ pub trait QueryScorer<TVector: ?Sized> {
 
     fn score_internal(&self, point_a: PointOffsetType, point_b: PointOffsetType) -> ScoreType;
 }
+
+pub fn score_multi<TMetric: Metric>(a: &MultiDenseVector, b: &MultiDenseVector) -> ScoreType {
```
Comment (agourlay) [inline lib/segment/src/vector_storage/query_scorer/mod.rs:20]: maybe you could have a comment on the function about the `MaxSim` operator and link towards a definition.

Comment (IvanPleshkov) [reply lib/segment/src/vector_storage/query_scorer/mod.rs]: added a comment with link to metric definition

File: lib/segment/src/vector_storage/query_scorer/mod.rs (hunk 1 above)
Comment (IvanPleshkov) [reply lib/segment/src/vector_storage/query_scorer/mod.rs:31]: renamed

**Review by agourlay — APPROVED** [review]

**Pull request merged**